contracts/pallet-ibc/src/lib.rs

```diff
@@ -724,6 +817,95 @@ pub mod pallet {
 				return Err(Error::<T>::InvalidTimestamp.into())
 			}
 
+			let mut ctx = Context::<T>::default();
+			let channel_end = ctx
+				.channel_end(&(PortId::transfer(), source_channel))
+				.map_err(|_| Error::<T>::ChannelNotFound)?;
+
+			let destination_channel = channel_end
+				.counterparty()
+				.channel_id
+				.ok_or_else(|| Error::<T>::ChannelNotFound)?;
+
+			let whitelisted = <WhitelistChannelIds<T>>::contains_key((
+				source_channel.sequence(),
```

Comment:
do we need to call sequence here? shouldnt this channel ids?

Reply:
channel sequence is channel id